checker/src/org/checkerframework/checker/i18n/I18nAnnotatedTypeFactory.java

```diff
@@ -26,6 +27,12 @@ public I18nAnnotatedTypeFactory(BaseTypeChecker checker) {
         this.postInit();
     }
 
+    @Override
+    protected AnnotationClassLoader createAnnotationClassLoader() {
+        // I18nChecker does not use a class loader
```

Comment:
Just to be clear: this is only an optimization to not instantiate the `AnnotationClassLoader`, which would cause some initialization work.
All of that work would be wasted, because `createSupportedTypeQualifiers` below returns a fixed set of qualifiers.

So this could be simplified by not setting the `loader` field in the constructor at all. Then subclasses that override `createSupportedTypeQualifiers` would not incur an overhead, even without also having to override `createAnnotationClassLoader`.
Method `loadTypeAnnotationsFromQualDir` would first call `loadTypeAnnotationsFromQualDir` to set the `loader` field.
The three classes that do something with the loader would be the only ones that need to override this method.

What do you think?

Reply:
There's a few more classes that do use a loader, but yes I'll change it to instantiate a loader within `createSupportedTypeQualifiers`.